1.6/contrib/configuration/jobs/OpenShift Sample/config.xml

```diff
@@ -0,0 +1,158 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<project>
+  <actions/>
+  <description></description>
+  <keepDependencies>false</keepDependencies>
+  <properties>
+    <hudson.model.ParametersDefinitionProperty>
+      <parameterDefinitions>
+
+        <hudson.model.TextParameterDefinition>
+          <name>OPENSHIFT_API_URL</name>
+          <description>URL of the OpenShift api endpoint.</description>
+          <defaultValue></defaultValue>
+        </hudson.model.TextParameterDefinition>
+
+        <hudson.model.TextParameterDefinition>
+          <name>AUTH_TOKEN</name>
+          <description>Authentication token of an account/serviceaccount for accessing the project to run builds and tag images.  If you are running Jenkins in the same OpenShift deployment that it is communicating with and you have granted appropriate roles to the service account for the pod (normally "default"), you can leave this blank.</description>
+          <defaultValue></defaultValue>
+        </hudson.model.TextParameterDefinition>
+        
+        <hudson.model.TextParameterDefinition>
+          <name>PROJECT</name>
+          <description>The OpenShift project this job will access</description>
+          <defaultValue>test</defaultValue>
+        </hudson.model.TextParameterDefinition>
+        
+        <hudson.model.TextParameterDefinition>
+          <name>BUILD_CONFIG</name>
+          <description>The BuildConfig to trigger</description>
+          <defaultValue>frontend</defaultValue>
+        </hudson.model.TextParameterDefinition>
+
+        <hudson.model.TextParameterDefinition>
+          <name>SERVICE</name>
+          <description>The service to test after build completion</description>
+          <defaultValue>frontend</defaultValue>
+        </hudson.model.TextParameterDefinition>
+
+        <hudson.model.TextParameterDefinition>
+          <name>DEPLOYMENT_CONFIG</name>
+          <description>The name of the DeploymentConfig that deploys the output of the BuildConfig</description>
+          <defaultValue>frontend</defaultValue>
+        </hudson.model.TextParameterDefinition>
+
+      </parameterDefinitions>
+    </hudson.model.ParametersDefinitionProperty>
+  </properties>
+  <scm class="hudson.scm.NullSCM"/>
+  <canRoam>true</canRoam>
+  <disabled>false</disabled>
+  <blockBuildWhenDownstreamBuilding>false</blockBuildWhenDownstreamBuilding>
+  <blockBuildWhenUpstreamBuilding>false</blockBuildWhenUpstreamBuilding>
+  <triggers/>
+  <concurrentBuild>false</concurrentBuild>
+  <builders>
+    <hudson.tasks.Shell>
+      <command>
+if [ -z &quot;$AUTH_TOKEN&quot; ]; then
+  AUTH_TOKEN=`cat /var/run/secrets/kubernetes.io/serviceaccount/token`
+fi
+
+if [ -e /run/secrets/kubernetes.io/serviceaccount/ca.crt ]; then
+  alias oc=&quot;oc --token=$AUTH_TOKEN --server=$OPENSHIFT_API_URL --certificate-authority=/run/secrets/kubernetes.io/serviceaccount/ca.crt &quot;
```

Comment:
I'm not sure how logging in would interact if you had multiple instances of the job running against different projects/servers... though `oc project` is also a stateful change that wouldn't play nicely with multiple jobs running against different projects